### PR TITLE
Add a link to download annotations

### LINKED ISSUE #323: Add download annotation button

### PR DESCRIPTION
**Pull request by jbeezley**
This functional, however, there is a problem when attempting to download a large annotation.  It takes the server several seconds to respond while the user has no indication that anything is happening.  The only way I can think of to deal with this is:
1. Add a `click` handler to the link which does the download and adds a spinner.  This will not work if the user tries to download via "Save As" in the context menu.
2. Add an endpoint `GET annotation/:id/download` that initiates a multipart download via paged responses.
If 2. is possible, I think it would be best because it doesn't override the browser's default behavior.

Closes #323

### CODE REVIEW

File: web_client/panels/AnnotationSelector.js [hunk 1]
```diff
@@ -13,7 +13,8 @@ import '../stylesheets/panels/annotationSelector.styl';
 var AnnotationSelector = Panel.extend({
     events: _.extend(Panel.prototype.events, {
         'click .h-toggle-annotation': 'toggleAnnotation',
-        'click .h-delete-annotation': 'deleteAnnotation'
+        'click .h-delete-annotation': 'deleteAnnotation',
+        'click .h-download-annotation': 'downloadAnnotation'
```
Comment (manthey) [inline web_client/panels/AnnotationSelector.js]: I'm failing to understand how this works -- where is `downloadAnnotation` defined?

**Comment by manthey** [comment]
I think we'll want a custom download endpoint.

Should we just be returning the `annotation` entry from this, as that is what gets uploaded?  The other top-level fields include potentially useful metadata (`creatorId`, `created`), plus some we may not want to return (`_elemenQuery`, `_version`) .

File: web_client/panels/AnnotationSelector.js (hunk 1 above)
Comment (jbeezley) [inline web_client/panels/AnnotationSelector.js]: That's leftover code.  I'll remove it.

**Comment by jbeezley** [comment]
I'm not sure.  My first instinct is to say just the annotation object, but I don't think it hurts to include the toplevel metadata as well.

**Comment by manthey** [comment]
Do either @cooperlab and @dgutman have an opinion on what should be downloaded when you download an annotation?  If including the metadata is desired, then this works now.

**Comment by cooperlab** [comment]
If we are restricting to human annotations then I would prefer including metadata (and any other info). Is small and can be easily discarded if not wanted.

**Pull request merged**